wordpress-mobile/WordPress-Android
# [Bloganuary] Show prompts bloganuary attribution UI

Fixes #19592 

When the prompts for today contains `bloganuary` in the `attribution` field we should show the proper icon and text in the attribution UI (the same we use for Day One).

**Note:** This still does not implement the updated UI, just reuses the attribution logic for the card.

![image](https://github.com/wordpress-mobile/WordPress-Android/assets/5091503/87103df0-9053-45f1-9ed9-dd808c970461)

## To test
The testing setup is the same as the one explained here: https://github.com/wordpress-mobile/WordPress-Android/pull/19590 (Charles Setup and mocked json response)

### Test Steps
1. Download, install, and log in the Jetpack version from this PR
3. **Disable** the `Map Local` setting in Charles (so we hit the actual endpoint)
4. Refresh the app dashboard
5. **Verify** the prompt card either shows no attribution UI or the `Day One` attribution (if today's prompt is a Day One prompt)
6. **Enable** the `Map Local` setting in Charles (so we hit our mocked json response)
7. Refresh the app dashboard
8. **Verify** the prompt card shows the attribution UI for `Bloganuary`

## Regression Notes
1. Potential unintended areas of impact
Incorrect attribution UI being shown.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual and Unit testing.

3. What automated tests I added (or what prevented me from doing so)
Test for the possible attribution types and proper mapping.

PR submission checklist:


UI Changes testing checklist:
